nxviz/plots.py

```diff
@@ -853,8 +853,7 @@ def draw_legend(self):
             x in seen or seen.add(x))]
 
         # Gets group labels
-        labels_group = sorted(list(set(
-            [self.graph.node[n][self.node_color] for n in self.nodes])))
+        labels_group = sorted(list(set([list(self.graph.nodes.values())[n][self.node_color] for n in np.arange(len(self.nodes))])))
```

Comment:
I think we can take the chance to simplify the implementation a bit, matching the NetworkX API.

Here's a suggestion, let me know if this works for you (to try it out, you'll have to "accept" the suggestion and then do a `git pull` on your local machine to pull down the changes):

```suggestion
    labels_group = sorted(
        set(
            [
                self.graph.nodes[n][self.node_color] for n in self.nodes
            ]
        )
    )
```

A few reasons why I think this might be a bit cleaner:

1. No indirection with `self.graph.nodes.values()`; we directly loop over the `self.graph`'s `.nodes` attribute.
1. The key problem, I think, was the use of `self.node` rather than `self.nodes`, for which the former doesn't work in NetworkX 2.4.
1. Removed an extraneous list (I think I wrote that line when I was not so familiar with Python's `sorted` semantics.

At the same time, I took the liberty of expanding out the brackets, so that the brackets are a little less intimidating to read.